javascript/tables/src/validationUtils.ts

```diff
@@ -0,0 +1,324 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type { ValidationTestResult } from '@datashaper/schema'
+import { DataType, ErrorCode } from '@datashaper/schema'
+
+export function validateRequiredConstraint(): (
+	values: unknown,
+	includeInstances: boolean,
+) => ValidationTestResult {
+	return function validateRequiredConstraint(
+		values: unknown,
+		includeInstances: boolean,
+	): ValidationTestResult {
+		const valuesCasted = values as string[]
+
+		if (!includeInstances) {
+			return {
+				fail: !valuesCasted.every(validateRequired),
+				indexes: [],
+				rule: ErrorCode.Required,
+			}
+		} else {
+			const resultIndexes: number[] = []
+
+			valuesCasted.map((value: string, index: number) => {
+				if (!validateRequired(value)) resultIndexes.push(index)
+			})
+
+			return {
+				fail: resultIndexes.length != 0,
+				indexes: resultIndexes,
+				rule: ErrorCode.Required,
+			}
+		}
+	}
+}
+
+export function validateRequired(element: string): boolean {
+	if (element == null || element == undefined || element.length == 0)
```

Comment:
This function can be simplified to just `return (element === null || element === undefined || element.length === 0)`

Since you are returning a boolean either way, you can just return the result of the evaluation. Actually you need to invert the logic in this case, since you were returning false:

`return (element !== null && element !== undefined && element.length !== 0)`

Also note the extra `=` for strict checks.

One more question on this: it should apply for any data type, right, not just strings?

You could handle that by specifying the type as unknown and just doing the null/undefined check and ignoring the length. I think zero-length strings should be handled by the type application utility anyway (i.e., they will be turned to null already).